llvm/docs/DTLTO.rst

```diff
@@ -60,165 +80,126 @@ Distributors are programs responsible for:
+  individual summary index file.
+- The first entry in each job's ``outputs`` array is the primary output object
+  file.
 
-- **Command-Line Template**: Command-line options are stored in a common
-  template to avoid duplication for each job. The template consists of an array
-  of strings and arrays. The arrays are placeholders which reference per-job
-  paths. This allows the remote compiler and its arguments to be changed without
-  updating the distributors.
+An array-of-strings representation is used for command-line arguments to avoid
+issues with quoting and shell escaping.
 
-Command-Line Expansion Example
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Command-line arguments and input/output files are stored separately to allow
+the remote compiler to be changed without updating the distributors, as the
+distributors do not need to understand the details of the compiler command
+line.
 
-To create the backend compilation commands, the command-line template is
-expanded for each job. Placeholders are expanded in the following way: The first
-array element specifies the name of the array field to look in. The remaining
-elements are converted to strings and concatenated. Integers are converted by
-indexing into the specified array.
+To generate the backend compilation commands, the job-specific and common
+arguments are combined.
 
-The example above generates the following backend compilation command for
-`main.o`:
+The above example generates the following backend compilation commands:
 
 .. code-block:: console
 
-    /usr/local/clang -O3 -fprofile-sample-use=my.profdata \
-        -o dtlto.1.51232.native.o -c -x ir dtlto.o \
-        -fthinlto-index=dtlto.1.51232.native.o.thinlto.bc --target=x86_64-sie-ps5
+    clang.exe -c -flto -fprofile-sample-use=my.prof t1.bc \
+        -fthinlto-index=t1.1.7944.native.o.thinlto.bc -o t1.1.7944.native.o \
+        -fproc-stat-report=t1.1.7944.stats.txt
 
-This expansion scheme allows the remote compiler to be changed without updating
-the distributors. For example, if the "args" field in the above example was
-replaced with:
+    clang.exe -c -flto -fprofile-sample-use=my.prof t2.bc \
+        -fthinlto-index=t2.2.7944.native.o.thinlto.bc -o t2.2.7944.native.o \
+        -fproc-stat-report=t2.2.7944.stats.txt
 
-.. code-block:: json
+Temporary Files
+---------------
 
-    "args": [
-        "custom-compiler",
-        "-opt-level=2",
-        "-profile-instrument-use-path=my.profdata",
-        "-output", ["primary_output", 0],
-        "-input", ["primary_input", 0],
-        "-thinlto-index", ["summary_index", 0],
-        "-triple", "x86_64-sie-ps5"
-    ]
+During its operation, DTLTO generates temporary files. Temporary files are
+created in the same directory as the linker's output file:
 
-Then distributors can expand the command line without needing to be updated:
+- **JSON Job Description File**:
 
-.. code-block:: console
+  - Format: ``<linker output file basename>.<PID>.dist-file.json``
```

Comment:
Do we really want to explicitly mention `PID` here because this is the cause of randomness that normally prevents DBS's from caching.

Reply:
I have mentioned this in the upstream review. I'm going to remove this section entirely.